File: plugins/bitcoin/bitcoin.5m.py
#!/usr/bin/env python3
"""Bitcoin Price Plugin - Uses Crossbar web API"""
import subprocess
import json

def crossbar_web(url):
    """Execute crossbar web command"""
    try:
        result = subprocess.run(['crossbar', 'web', url], capture_output=True, text=True, timeout=10)
        if result.returncode == 0:
            return result.stdout.strip()
    except Exception:
        pass
    return None

# Try Crossbar API first
response = crossbar_web('api.coinbase.com/v2/prices/BTC-USD/spot')

# Fallback to urllib
if not response:
    import urllib.request
    try:
        with urllib.request.urlopen('https://api.coinbase.com/v2/prices/BTC-USD/spot', timeout=10) as r:
            response = r.read().decode()
    except Exception:
        pass

if response:
    try:
        data = json.loads(response)
        price = data.get('data', {}).get('amount', '--')
        
        try:
            formatted = f"{float(price):,.0f}"
        except (ValueError, TypeError):
            formatted = price
        
        print(f"₿ ${formatted}")
        print("---")
        print(f"BTC/USD: ${price}")
        print("Source: Coinbase")
    except json.JSONDecodeError:
        print("₿ Parse Error")
else:
    print("₿ Error")
    print("---")
    print("Failed to fetch price")

print("---")
print("Refresh | refresh=true")

File: plugins/bitcoin/bitcoin.5m.js
#!/usr/bin/env node
/**
 * Bitcoin Price Plugin - Uses Crossbar web API
 */
const { execSync } = require('child_process');
const https = require('https');

function crossbarWeb(url) {
    try {
        return execSync(`crossbar web ${url}`, { encoding: 'utf8', timeout: 10000 }).trim();
    } catch {
        return null;
    }
}

// Try Crossbar API first
let response = crossbarWeb('api.coinbase.com/v2/prices/BTC-USD/spot');

if (response) {
    try {
        const data = JSON.parse(response);
        const price = data.data?.amount || '--';
        const formatted = parseFloat(price).toLocaleString('en-US', { maximumFractionDigits: 0 });
        
        console.log(`₿ $${formatted}`);
        console.log('---');
        console.log(`BTC/USD: $${price}`);
        console.log('Source: Coinbase');
        console.log('---');
        console.log('Refresh | refresh=true');
    } catch {
        console.log('₿ Parse Error');
    }
} else {
    // Fallback to native https
    https.get('https://api.coinbase.com/v2/prices/BTC-USD/spot', (res) => {
        let data = '';
        res.on('data', (chunk) => data += chunk);
        res.on('end', () => {
            try {
                const json = JSON.parse(data);
                const price = json.data?.amount || '--';
                const formatted = parseFloat(price).toLocaleString('en-US', { maximumFractionDigits: 0 });
                
                console.log(`₿ $${formatted}`);
                console.log('---');
                console.log(`BTC/USD: $${price}`);
                console.log('Source: Coinbase');
            } catch {
                console.log('₿ Parse Error');
            }
            console.log('---');
            console.log('Refresh | refresh=true');
        });
    }).on('error', () => {
        console.log('₿ Error');
        console.log('---');
        console.log('Failed to fetch price');
    });
}
